tests/testOpExportMultipageTiff.py

```diff
@@ -18,96 +17,71 @@
-        self._tmpdir = tempfile.mkdtemp()
+    def __getitem__(self, item: int) -> Operator:
+        return self._ops.__getitem__(item)
 
-        # Generate some test data
-        self.dataShape = (1, 10, 64, 128, 2)
-        self._axisorder = 'tzyxc'
-        self.testData = vigra.VigraArray( self.dataShape,
-                                         axistags=vigra.defaultAxistags(self._axisorder),
-                                         order='C' ).astype(numpy.uint8)
-        self.testData[...] = numpy.indices(self.dataShape).sum(0)
+    def close(self) -> None:
+        for op in reversed(self._ops):
+            op.cleanUp()
 
-    def tearDown(self):
-        # Clean up
-        shutil.rmtree(self._tmpdir)
-        
-    def test_basic(self):
-        opSource = OpArrayPiper(graph=self.graph)
-        opSource.Input.setValue( self.testData )
-        
-        opData = OpBlockedArrayCache( graph=self.graph )
-        opData.BlockShape.setValue( self.testData.shape )
-        opData.Input.connect( opSource.Output )
-        
-        filepath = os.path.join( self._tmpdir, 'multipage.tiff' )
-        logger.debug( "writing to: {}".format(filepath) )
-        
-        opExport = OpExportMultipageTiff(graph=self.graph)
-        opExport.Filepath.setValue( filepath )
-        opExport.Input.connect( opData.Output )
 
-        # Run the export
-        opExport.run_export()
+def test_OpExportMultipageTiff():
+    shape = 3, 10, 64, 128, 2
+    axes = "tzyxc"
+    dtype = numpy.uint32
+    data = numpy.arange(numpy.prod(shape), dtype=dtype).reshape(shape)
+    expected = vigra.VigraArray(data, axistags=vigra.defaultAxistags(axes), order="C")
 
-        opReader = OpTiffReader( graph=self.graph )
-        try:
-            opReader.Filepath.setValue( filepath )
-    
-            # Re-order before comparing
-            opReorderAxes = OpReorderAxes( graph=self.graph )
-            try:
-                opReorderAxes.AxisOrder.setValue( self._axisorder )
-                opReorderAxes.Input.connect( opReader.Output )
-                
-                readData = opReorderAxes.Output[:].wait()
-                logger.debug("Expected shape={}".format( self.testData.shape ) )
-                logger.debug("Read shape={}".format( readData.shape ) )
-                
-                assert opReorderAxes.Output.meta.shape == self.testData.shape, \
-                    "Exported files were of the wrong shape or number."
-                assert (opReorderAxes.Output[:].wait() == self.testData.view( numpy.ndarray )).all(), \
-                    "Exported data was not correct"
-            finally:
-                opReorderAxes.cleanUp()
-        finally:
-            opReader.cleanUp()
+    graph = Graph()
 
-if __name__ == "__main__":
-    # Run this file independently to see debug output.
-    logger.setLevel(logging.DEBUG)
-    logger.addHandler(logging.StreamHandler(sys.stdout))
+    with tempfile.NamedTemporaryFile(suffix=".tiff") as file:
```

Comment:
this fails on windows :/

from [the docs](https://docs.python.org/3.6/library/tempfile.html#tempfile.NamedTemporaryFile):
> This function operates exactly as TemporaryFile() does, except that the file is guaranteed to have a visible name in the file system (on Unix, the directory entry is not unlinked). That name can be retrieved from the name attribute of the returned file-like object. **Whether the name can be used to open the file a second time, while the named temporary file is still open, varies across platforms (it can be so used on Unix; it cannot on Windows NT or later)**. If delete is true (the default), the file is deleted as soon as it is closed. The returned object is always a file-like object whose file attribute is the underlying true file object. This file-like object can be used in a with statement, just like a normal file.

Reply:
Oops, missed it, sorry. Replaced it with the temporary directory (could use `mkstemp`, but contextmanagers are too awesome).